update_deps/propagate_updates.ps1

```diff
@@ -1,2 +1,509 @@
+    }
+}
+
+
+# set PR for Azure repo to merge automatically once build completes
+function set-autocomplete-azure {
+    param(
+        $pr_url,
+        $creator_id
+    )
+    $request_url = $pr_url + '?api-version=6.0'
+    $body =@{
+        autoCompleteSetBy=@{
+            id=$creator_id
+        }
+        completionOptions=@{
+            mergeStrategy="squash"
+        }
+    }
+    $body = $body | ConvertTo-Json
+    $response = Invoke-WebRequest -URI $request_url -UseBasicParsing -Method Patch -Headers $azure_header -Body $body -ContentType "application/json"
+    if(!$response) {
+        Write-Error "Failed to set autocomplete for PR $pr_url"
+        exit -1
+    }
+}
+
+
+# wait until build completes for Azure repo
+function wait-until-complete-azure {
+    param(
+        [string] $pr_url
+    )
+    $status = ""
+    while($true){
+        $response = Invoke-WebRequest -URI $pr_url -UseBasicParsing -Method Get -Headers $azure_header
+        if(!$response) {
+            Write-Error "Failed to get PR: $pr_url"
+            exit -1
+        }
+        $content =  $response.Content | ConvertFrom-Json
+        $status = $content.status
+        $merge_status = $content.mergeStatus
+        if($status -ne "active" -or $merge_status -ne "succeeded") {
+            break
+        }
+        spin 10
+    }
+    if($status -ne "completed") {
+        Write-Host "Problem with pull request: $pr_url"
+        Write-Error $response.Content
+        exit -1
+    }
+}
+
+
+# update dependencies for Azure repo
+function update-repo-azure {
+    param(
+        [string] $repo_name
+    )
+    Write-Host "`nCreating work item"
+    $work_item_id = create-work-item-azure
+    Write-Host "Creating PR"
+    $create_pr_response = create-pr-azure $repo_name
+    $pr_artifact_id = $create_pr_response.artifactId
+    Write-Host "Linking work item to PR"
+    link-work-item-to-pr-azure $work_item_id $pr_artifact_id
+    Write-Host "Approving PR"
+    approve-pr-azure $create_pr_response.url $create_pr_response.createdBy.id
+    Write-Host "Enabling PR to autocomplete"
+    set-autocomplete-azure $create_pr_response.url $create_pr_response.createdBy.id
+    Write-Host "Waiting for build to complete"
+    wait-until-complete-azure $create_pr_response.url
+}
+
+
+# determine whether given repo is an azure repo or a github repo
+function  get-repo-type {
+    param (
+        [string] $repo_name
+    )
+    cd $repo_name
+    $repo_url = git config --get remote.origin.url
+    cd ..
+    Write-Host $repo_url -NoNewline
+    if($repo_url.Contains("github")){
+        return "github"
+    }elseif ($repo_url.Contains("azure")) {
+        return "azure"
+    }
+    return "unknown"
+}
+
+
+# update dependencies for given repo
+function update-repo {
+    param(
+        [string] $repo_name
+    )
+    Write-Host "`n`nUpdating repo $repo_name"
+    [string]$git_output = (update-local-repo $repo_name)
+    if($git_output.Contains("nothing to commit")) {
```

Comment:
I love how you broke down the code into functions and how organized it is

Reply:
Thanks!